lib/pages/community.dart

```diff
@@ -541,3 +522,91 @@ class _Divider extends StatelessWidget {
         child: Divider(),
       );
 }
+
+class _FollowButton extends HookWidget {
+  final CommunityView community;
+
+  _FollowButton(this.community);
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final colorOnTopOfAccent = textColorBasedOnBackground(theme.accentColor);
+    final subscribed = community.subscribed ?? false;
+    final token = tokenOrNull(context, community.instanceUrl);
+
+    final isSubbed = useState(subscribed);
+    final showSpinner = useState(false);
+    final isPending = useState(false);
+
+    subscribe() async {
+      if (token == null) {
+        Scaffold.of(context).showSnackBar(
+            SnackBar(content: Text("can't sub when you're not logged in")));
+        return;
+      }
+
+      isPending.value = true;
+      var spinnerTimer =
+          Timer(Duration(milliseconds: 500), () => showSpinner.value = true);
+
+      final api = LemmyApi(community.instanceUrl).v1;
+      try {
+        await api.followCommunity(
+            communityId: community.id, follow: !isSubbed.value, auth: token);
+        isSubbed.value = !isSubbed.value;
+        // ignore: avoid_catches_without_on_clauses
+      } catch (e) {
+        Scaffold.of(context).showSnackBar(SnackBar(
+            content: Row(
+          children: [
+            Icon(Icons.warning),
+            SizedBox(width: 10),
+            Text("couldn't ${isSubbed.value ? 'un' : ''}sub :<"),
+          ],
+        )));
```

Comment:
please add one trailing comma, this formatting makes me anxious

Reply:
lmao